test/support/factories/keyword_factory.ex

```diff
@@ -0,0 +1,29 @@
+defmodule GoogleScraping.KeywordFactory do
+  alias GoogleScraping.Dashboard.Schemas.Keyword
+
+  defmacro __using__(_opts) do
+    quote do
+      @fixture_path "test/support/fixtures/files"
+
+      def keyword_factory do
+        user = insert(:user)
+
+        %Keyword{
+          name: Faker.Lorem.word(),
+          user_id: user.id,
```

Comment:
```suggestion
           user: build(:user)
```

Reply:
With that, it will be impossible to specify a user for the keyword, like `insert(:keyword, user: user)`, it is just doesn't accept provided user without any errors, so I used here `user_id` which is working fine.

I'm not sure what exactly is not working here.

Reply:
It fails due to it is not loading `user` automatically.

It can be solved by preloading user like
```ruby
  def list_keywords(user_id) do
    KeywordQuery.user_keywords_query(user_id)
    |> preload(:user)
    |> Repo.all()
  end
```

Or changing test like
```
      keywords= Keywords.list_keywords(user.id

      assert length(keywords) == 1
      assert List.first(keywords).id == keyword.id
```

Or just using the current one(`user_id`) in the factory.

I am fine with any option.